internal/storage/storage_handle_test.go

```diff
@@ -66,19 +65,15 @@ func (t *StorageHandleTest) invokeAndVerifyStorageHandle(sc StorageClientConfig)
 }
 
 func (t *StorageHandleTest) TestBucketHandleWhenBucketExists() {
-	fakeClient := t.fakeStorageServer.Client()
-	storageClient := &storageClient{client: fakeClient}
-
+	storageClient := t.fakeStorage.CreateStorageHandle()
```

Comment:
rename the variable to storageHandle. You are creating storageHandle instance here not storageClient. 
Same for below test as well.

Reply:
fixed